test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/SemaphoreTest.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.concurrent
+
+import java.util.concurrent.{Semaphore, TimeUnit}
+import org.junit.Assert._
+import org.junit.Test
+
+class SemaphoreTest {
+
+  @Test def `default constructor should be unfair`(): Unit = {
```

Comment:
Please chose a test naming style that is consistent with recent PRs (e.g. #4299). There is more discussion about this in #4266.